### PR TITLE
Use latest ILC to AOT-compile ILC

### PR DESCRIPTION
**Pull request by sbomer**
We need to build with a recent enough ILC that has the fix from https://github.com/dotnet/runtime/pull/87785. This sets up a dependency on the runtime -> runtime package flow to build using the latest ILC package.

### CODE REVIEW

**Review by LakshanF — APPROVED** [review]
Thanks! I validated that the NativeAOT ILC build with these changes (`artifacts\bin\coreclr\windows.x64.Debug\ilc-published`) does not cause problems.

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -314,6 +314,10 @@
       <Uri>https://github.com/dotnet/runtime</Uri>
       <Sha>5b4bbda1c1c8a5cc4ced9facdacb4e531dfc8b3c</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.DotNet.ILCompiler" Version="">
```
Comment (jkoritzinsky) [inline eng/Version.Details.xml:317]:
```suggestion
    <Dependency Name="Microsoft.DotNet.ILCompiler" Version="8.0.0-rc.1.23374.1">
```

**Review by jkotas — APPROVED** [review]
Is this a permanent change that is here to stay, or a temporary change that will be reverted after next SDK update?

**Comment by jkotas** [comment]
> I'm thinking we could make it permanent, so that future ILCompiler fixes flow through the system quickly - do you have an opinion?

We can give it a try. This setup may break when we need to push through a change that makes current native AOT incompatible with older SDK.

**Comment by am11** [comment]
> I'm thinking we could make it permanent

Can we take a step further and bring it on the same plan as crossgen2? i.e. use "live build" of (unpublished) ilc to publish ilc for shipping nuget package. Then use the published bits in runtime and libraries testing as we do today with ilc (but not with crossgen2 yet due to issues on arm32 platform).

**Comment by MichalStrehovsky** [comment]
> > I'm thinking we could make it permanent
> 
> Can we take a step further and bring it on the same plan as crossgen2? i.e. use "live build" of (unpublished) ilc to publish ilc for shipping nuget package. Then use the published bits in runtime and libraries testing as we do today with ilc (but not with crossgen2 yet due to issues on arm32 platform).

We scrapped the underlying tracking issue due to funding: https://github.com/dotnet/runtime/issues/67742#issuecomment-1601846167. It would be nicer to do it that way as it would avoid any versioning issue, but the LKG approach also has an advantage (we do all testing with the exact configuration that we're shipping).

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Comment by jkotas** [comment]
> I would suggest moving PR builds to macOS 12 as well. Are we ok with requiring XCode 14 to build NativeAot apps on mac?

I do not see a problem with that.

**Pull request merged**